src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -153,31 +158,54 @@ class ComService(
     }
   }
 
-  private fun findPrisonersForUnstartedRecords(record: List<Pair<CaseloadResult, Licence?>>): Map<String, PrisonerSearchPrisoner> {
-    val prisonNumbers = record
-      .filter { (_, licence) -> licence == null }
-      .mapNotNull { (result, _) -> result.identifiers.noms }
-
-    val prisoners = this.prisonerSearchApiClient.searchPrisonersByNomisIds(prisonNumbers)
+  private fun findPrisonersWithoutDraftLicences(
+    eligiblePrisonersList: List<PrisonerSearchPrisoner>,
+    record: List<Pair<CaseloadResult, Licence?>>,
+  ): Map<String, PrisonerSearchPrisoner> {
+    val eligiblePrisoners =
+      eligiblePrisonersList.filter { prisoner ->
+        !record.isLicenceInGivenStatusPresent(prisoner.prisonerNumber, LicenceStatus.DRAFT_LICENCES) ||
+          record.isLicenceInGivenStatusPresent(prisoner.prisonerNumber, LicenceStatus.ON_PROBATION_STATUSES.toList())
+      }
 
-    if (prisoners.isEmpty()) {
+    if (eligiblePrisoners.isEmpty()) {
       return emptyMap()
     }
 
-    val initialCvlEligiblePrisoners = prisoners
-      .filter {
-        eligibilityService.isEligibleForCvl(it)
-      }
-
-    val prisonerBookingsWithHdc = initialCvlEligiblePrisoners.findBookingsWithHdc()
+    val prisonerBookingsWithHdc = eligiblePrisoners.findBookingsWithHdc()
 
     val eligibleForCvlPrisoners =
-      initialCvlEligiblePrisoners.filterNot { prisonerBookingsWithHdc.contains(it.bookingId.toLong()) }
+      eligiblePrisoners.filterNot { prisonerBookingsWithHdc.contains(it.bookingId.toLong()) }
 
     return eligibleForCvlPrisoners.associateBy { it.prisonerNumber }
   }
 
-  private fun CaseloadResult.createUnstartedRecord(
+  private fun List<Pair<CaseloadResult, Licence?>>.isLicenceInGivenStatusPresent(
+    prisonerNumber: String,
+    status: List<LicenceStatus>,
+  ) =
+    any { (probationSearchResult, licence) ->
+      probationSearchResult.identifiers.noms == prisonerNumber &&
+        licence != null && licence.statusCode in status
+    }
+
+  private fun findPrisonersEligibleToCreateLicence(record: List<Pair<CaseloadResult, Licence?>>): List<PrisonerSearchPrisoner> {
+    val prisonNumbers = record
+      .mapNotNull { (result, _) -> result.identifiers.noms }
+
+    val prisoners = this.prisonerSearchApiClient.searchPrisonersByNomisIds(prisonNumbers)
+      .filter { prisoner ->
+        eligibilityService.isEligibleForCvl(prisoner) ||
+          record.isLicenceInGivenStatusPresent(prisoner.prisonerNumber, listOf(LicenceStatus.ACTIVE))
+      }
+
+    return prisoners
+  }
+
+  private fun filterPrisonersWithNonDraftLicences(record: List<Pair<CaseloadResult, Licence?>>): List<Pair<CaseloadResult, Licence?>> =
+    record.filter { (_, licence) -> licence == null || licence.statusCode in LicenceStatus.DRAFT_LICENCES || licence.statusCode in LicenceStatus.ON_PROBATION_STATUSES.toList() }
```

Comment:
This check is duplicated as we are already checking "DRAFT_LICENCES" and "ON_PROBATION_STATUSES" previously?